common/types.go

```diff
@@ -373,11 +373,37 @@ func (a Address) Value() (driver.Value, error) {
 	return a[:], nil
 }
 
-// Checks if an address is a valid account in our node's sharded address space
+// IsInChainScope checks if an address is a valid account in our node's sharded address space
 func (a Address) IsInChainScope() bool {
 	return NodeLocation.ContainsAddress(a)
 }
 
+// Location looks up the chain location which contains this address
+func (a Address) Location() *Location {
+	// Search zone chain address space
+	for r := 0; r < NumRegionsInPrime; r++ {
+		for z := 0; z < NumZonesInRegion; z++ {
+			l := Location{byte(r), byte(z)}
+			if l.ContainsAddress(a) {
+				return &l
+			}
+		}
+	}
+	// Search region chain address space
```

Comment:
It could, but I intentionally left them separate for two reasons:
1) zone addresses will be more common, so I want to prioritize searching them first
2) separating the two felt clearer -- when reading the code, you can clearly see the three groups that are being searched.

Reply:
zone addresses maybe more common but it is an n*m check vs a single check.

Reply:
I can combine them

Reply:
done: 2708aff